Resources/Computing/sudoComputingElement.py

```diff
@@ -0,0 +1,182 @@
+""" A computing element class that uses sudo
+"""
+
+__RCSID__ = "$Id$"
+
+import os
+import pwd
+import stat
+import distutils.spawn
+
+import DIRAC
+
+from DIRAC                                                  import S_OK, S_ERROR
+
+from DIRAC.ConfigurationSystem.Client.Config                import gConfig
+from DIRAC.Resources.Computing.ComputingElement             import ComputingElement
+from DIRAC.Core.Utilities.ThreadScheduler                   import gThreadScheduler
+from DIRAC.Core.Utilities.Subprocess                        import shellCall
+
+MandatoryParameters = [ ]
+
+class sudoComputingElement( ComputingElement ):
+
+  mandatoryParameters = MandatoryParameters
+
+  #############################################################################
+  def __init__( self, ceUniqueID ):
+    """ Standard constructor.
+    """
+    ComputingElement.__init__( self, ceUniqueID )
+    self.submittedJobs = 0
+
+  #############################################################################
+  def _addCEConfigDefaults( self ):
+    """Method to make sure all necessary Configuration Parameters are defined
+    """
+    # First assure that any global parameters are loaded
+    ComputingElement._addCEConfigDefaults( self )
+    # Now sudo CE specific ones
+
+  #############################################################################
+  def submitJob( self, executableFile, proxy, dummy = None ):
+    """ Method to submit job, overridden from super-class.
+    """
+    self.log.verbose( 'Setting up proxy for payload' )
+    result = self.writeProxyToFile( proxy )
+    if not result['OK']:
+      return result
+
+    payloadProxy = result['Value']
+    if not os.environ.has_key( 'X509_USER_PROXY' ):
+      self.log.error( 'X509_USER_PROXY variable for pilot proxy not found in local environment' )
+      return S_ERROR( 'X509_USER_PROXY not found' )
+
+    pilotProxy = os.environ['X509_USER_PROXY']
+    self.log.info( 'Pilot proxy X509_USER_PROXY=%s' % pilotProxy )
+
+    # First username in the sequence to use when running payload job
+    # If first is pltXXp00 then have pltXXp01, pltXXp02, ...
+    try:
+      baseUsername = gConfig.getValue( "/LocalSite/SudoBaseUsername", "" )
```

Comment:
ComputingElement configuration should be taken from the usual place as for any CEs and obtained in the code from the ceParameters variable